src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -48,6 +48,56 @@ static class BitOperations
             19, 27, 23, 06, 26, 05, 04, 31
         };
 
+        /// <summary>
+        /// Evaluate whether a given integral value is a power of 2.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsPow2(int value) => (value & (value - 1)) == 0 && value >= 0;
+
+        /// <summary>
+        /// Evaluate whether a given integral value is a power of 2.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [CLSCompliant(false)]
+        public static bool IsPow2(uint value)
```

Comment:
Can you also update:
https://github.com/dotnet/runtime/blob/64da821c7b3e60f9aeaf2c3d544d33d19e2fc82e/src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs#L224
and
https://github.com/dotnet/runtime/blob/64da821c7b3e60f9aeaf2c3d544d33d19e2fc82e/src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs#L237
to use `IsPow2`?

Reply:
We may want to use just `if ((value & (value - 1)) != 0)` without checking for 0. Otherwise `Log2Ceiling(0)` would return `1`, which is greater than `Log2Ceiling(1)`.

Reply:
Log2Ceiling(0) doesn't matter right now because no code will do that.  If Log2Ceiling is every made public, erroneous inputs will need to be considered then.

Reply:
Not only it would be correct, it should be faster as well. Ideally `(value & (value - 1))` should be compiled into a single `bslr` instruction on x64.

Reply:
Whatever implementation is chosen for IsPow2, it doesn't change the fact that these call sites should use IsPow2.